app/api/routes.py

```diff
@@ -179,6 +182,57 @@ def get_resources():
     return standardize_response(payload=dict(data=resource_list, **pagination_details))
 
 
+def search(term):
+    query = Resource.query
+    page = 0
+
+    if request.args.get('page') is not None:
+        page = int(request.args.get('page'))
+
+    indicies = client.list_indices()
+    for ind in indicies['items']:
+        if ind['name'] == 'resources_api':
+            curr_index = ind
+
+            if curr_index['entries'] != query.count():
+                print("ReIndexing Database")
+
+                index.clear_objects()
+                db_list = [u.serialize for u in query.all()]
+                index.save_objects(db_list)
+
+            break
```

Comment:
From 192: 204
This block indexes the db to algolia if the database and their index are different sizes. 
I tried to include this code in multiple places where it seemed appropriate, but I was running into import errors trying to get the Query to run. 

I might just have a function that fires it off without checking on startup.

Reply:
This is the inappropriate place for this. 

Imagine the front-end calls the search end point AFTER a new item is added.

The request will now take a much longer time and will return unexpected results.

The Algolia index should be updated specifically when resources are created, deleted, or updated. This will lead to more code, but it is more testable/expected

Reply:
Yeah, I dont want this code to be here, I just kinda left it so that if someone runs this code, it will at least populate their index. 

I attempted to have it only run on startup for now (within app.create_app) but I kept having import errors when I would import Resources. I also tried having it run within cli.register but that was a similar situation.